mobile-browser-based-version/src/components/building_frames/csv/CsvTrainingFrame.vue

```diff
@@ -0,0 +1,138 @@
+<template>
+  <TrainingFrame
+    :Id="Id"
+    :Task="Task"
+    :dataPreprocessing="dataPreprocessing"
+    :num_peers="num_peers"
+    :nbrClasses="1"
+    :precheckData="precheckData"
+  >
+    <template v-slot:dataExample>
+      <!-- Data Point Example -->
+      <div class="relative p-4 overflow-x-scroll">
+        <table class="table-auto">
+          <thead>
+            <tr>
+              <th
+                v-for="example in dataExample"
+                :key="example"
+                class="px-4 py-2 text-emerald-600"
+              >
+                {{ example.columnName }}
+              </th>
+            </tr>
+          </thead>
+          <tbody>
+            <tr>
+              <td
+                v-for="example in dataExample"
+                :key="example"
+                class="border border-emerald-500 px-4 py-2 text-emerald-600 font-medium"
+              >
+                {{ example.columnData }}
+              </td>
+            </tr>
+          </tbody>
+        </table>
+      </div>
+    </template>
+    <template v-slot:extra>
+      <!-- Modification of Header Card -->
+      <icon-card
+        header="Map My Data"
+        description="If the header of the file that you've uploaded differs from the one shown in example, you can map the expected header to your header format bellow."
+      >
+        <template v-slot:icon><bezier-2 /></template>
+        <template v-slot:extra>
+          <!-- Display all the possible headers -->
+          <div id="mapHeader">
+            <ul
+              class="grid grid-cols-1 gap-4 p-4 lg:grid-cols-2 xl:grid-cols-4"
+            >
+              <li
+                class="border-gray-400"
+                v-for="header in headers"
+                :key="header.id"
+              >
+                <div
+                  class="select-none p-2 transition duration-500 ease-in-out transform hover:-translate-y-2 rounded-2xl border-2 p-6 hover:shadow-2xl border-primary-dark"
+                >
+                  <div class="grid grid-cols-3 items-center p-2">
+                    <div class="pl-1">
+                      <div class="font-medium">
+                        <div class="flex flex-row justify-start">
```

Comment:
Is it maybe better to combine class attributes into a single div element? (Each new element adds x2 lines of code).

e.g.

<div class="grid grid-cols-3 items-center p-2 pl-1 font-medium flex flex-row justify-start">
</div>